lib/matplotlib/figure.py

```diff
@@ -938,6 +938,46 @@ def _break_share_link(ax, grouper):
         # Break link between any twinned axes
         _break_share_link(ax, ax._twinned_axes)
 
+    def clf(self, keep_observers=False):
+        """
+        Clear the figure.
+
+        Set *keep_observers* to True if, for example,
+        a gui widget is tracking the Axes in the figure.
+        """
+        self.suppressComposite = None
+        self.callbacks = cbook.CallbackRegistry()
+
+        for subfig in self.subfigs: 
+            #first clear the axes in any subfigures
+            subfig.clf(keep_observers=keep_observers)
+
+        for ax in tuple(self.axes):  # Iterate over the copy.
+            ax.cla()
+            self.delaxes(ax)  # Remove ax from self._axstack.
+
+        toolbar = getattr(self.canvas, 'toolbar', None)
+        if toolbar is not None:
+            toolbar.update()
+        self.artists = []
+        self.subfigs = [sfig for sfig in self.subfigs if sfig.axes != []]
```

Comment:
is there any reason for this list to change?

Reply:
I wasn't sure if maybe a subfigure had a GUI widget tracking one of its axes maybe we needed to keep the subfigure around.

Reply:
I removed this code, but unsure if this could introduce issues with callbacks. Not quite 100% on how they work.